addons/mail/static/src/models/thread/thread.js

```diff
@@ -831,6 +839,20 @@ function factory(dependencies) {
             return this.messaging.currentPartner.moderatedChannelIds.includes(this.id);
         }
 
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsPinned() {
+            // The channel is not managed by manual subscription
+            // but by res.groups membership. The server returned this channel,
+            // so it is mandatorily pinned
+            if (this.group_based_subscription) {
+                return true;
+            }
```

Comment:
Seems off to have `isPinned` field managing this case.
I would say it's the UI that shouldn't allow unpinning such channels.

This eases removing this constraint in future, and also could prevent some bugs right now, like enforcing a `pinned` on UI but it still allow a "leave" command that makes a RPC and could unsubscribe the current partner from channel. Basically this would either show buggy state or it's too coupled with server implementation.

Reply:
There is a lot here that I have no control of in a bug fix setting
the UI allready prevents unpinning those channels by not displaying the leave button (SidebarItem XML)

If I remove the res.group from the channel, I don't expect anything in particular because the write on mail.channel won't trigger a bus notification (`_subscribe_users
` in mail_channel.py). Only a reload would show the changes.

Those channels are displayed, in JS terms, they are pinned, regardless of what the server sends.

I agree with you on the conclusion of that discussion: there is a conceptual problem in how the design is made in python, but I think it is out of scope.

Reply:
> If I remove the res.group from the channel, I don't expect anything in particular because the write on mail.channel won't trigger a bus notification (`_subscribe_users ` in mail_channel.py). Only a reload would show the changes.
> 
> Those channels are displayed, in JS terms, they are pinned, regardless of what the server sends.
> 
> I agree with you on the conclusion of that discussion: there is a conceptual problem in how the design is made in python, but I think it is out of scope.

Yes, the problems of making changes in `mail.channel` form view and no visible changes in discuss/messaging menu/etc. is a big problem since the availability of accessing `mail.channel` form view (IIRC, 12.0 as "bug-fixing" by xdo).

> the UI allready prevents unpinning those channels by not displaying the leave button (SidebarItem XML)

Ok, so I would be fine to just keep that, and remove dependency on `group_based_subscription` in computation of `isPinned`, because I don't see the purpose of that.

Reply:
Btw we already had a comments above in the PR about this.

The main issue is that the server can inconsistently send is_pinned=false on channels with group_based_subscription set, and we want those to be pinned no matter what.

Maybe fix is appropriate at convertData/rpc return level instead? But we can't rely on received data for sure.

Reply:
(Sorry if comments look rude, that's not my intent at all. if I miss something super important that requires this, don't hesitate to let me know ;) )

Reply:
> Btw we already had comments above in the PR about this.
> 
> The main issue is that the server can inconsistently send is_pinned=false on channels with group_based_subscription set, and we want those to be pinned no matter what.
> 
> Maybe fix is appropriate at convertData/rpc return level instead? But we can't rely on received data for sure.

Oh ok, that makes way more sense with this piece of information!

I'm not entirely sure the extend of complications with channel_info providing `is_pinned=false` with `group_based_subscription=true`, but I guess it has to implementation detail of `channel.partner` relational table.

I would prefer auto-setting `isPinned` in `mail.thread:convertData` when `group_based_subscription` is set. This put technical constraint at minimum.

Reply:
@alexkuhn 
what do you mean auto-setting isPinned in convertData ?
you want to assign a value to the computed field ?
In that case any subsequent re-compute will override that forced value, I'm pretty sure we don't want that.

Or did you mean assigning serverPinState when group_based... is set ?
In that case, serverPinState won't be the server pin state anymore ....

Reply:
I think we are wasting our time and should fix the python instead?

`group_ids` help text states that
"Members of those groups will automatically added as followers. "
             "Note that they will be able to manage their subscription manually "
             "if necessary."

So for me, that's a sign people should be able to leave those just like normal channels.
"auto subscribe" should just do that, automatically subscribe, and nothing else.
When that happens, is_pinned should be set to true (in `_subscribe_users`).

Rest of the code should not have to worry about this option anywhere else.

Reply:
If we can just adapt python so that provided `is_pinned` is consistent on `group_based_subscription` channels, then I'm all for this solution over hacks in JS with `isPinned`.

Reply:
> Ok, so I would be fine to just keep that, and remove dependency on group_based_subscription in computation of isPinned, because I don't see the purpose of that.
There is no other purpose than putting as dependency a field that will be useful for computation, for consistency. The field definition is its own documentation...

Reply:
> > Ok, so I would be fine to just keep that, and remove dependency on group_based_subscription in computation of isPinned, because I don't see the purpose of that.
> 
> There is no other purpose than putting as dependency a field that will be useful for computation, for consistency. The field definition is its own documentation...

Yes, but here I meant the whole logic of having `isPinned` value having a special case with `group_based_subscription`. In this conversation, we lead to making python changes to that `is_pinned` in `channel_info` is correct. If we can do this, then there's no more reason to have such strange logic (even should allow leaving a group based subscription channel)

Reply:
I am not very "chaud patate" to change the python:
- The least we can say is that what exists and what we want both are very blurry
- None of those "features" is tested in python
- If we are to change the python, we should re-think the way pinned work/is
- there are migration too to think of, because those group based channel won't be is_pinned on an existing db

Of course, all of those are overcomable, as part of clear-cut task, not as a fix.
Then again, the behavior that should be aimed at is the one before the refactoring, it has some sense anyway, and the one we want to give it is not clear enough.

All in all, I think the risks of changing the python outweigh the benefits of not having some mildly odd stuff in js.

Reply:
Ok to consider out of scope, but should create new task and add fixme comment at this condition then.

Reply:
But do it in convert data.

> you want to assign a value to the computed field ?
> In that case any subsequent re-compute will override that forced value, I'm pretty sure we don't want that.

You would not write it on the computed field isPinned, you would write it on serverPinState.

Reply:
> But do it in convert data.
> 
> > you want to assign a value to the computed field ?
> > In that case any subsequent re-compute will override that forced value, I'm pretty sure we don't want that.
> 
> You would not write it on the computed field isPinned, you would write it on serverPinState.

Yes, it's nice to keep weirdness of handling of server formatter in `convertData`. This prevents having hacky logic everywhere in JS.